src/lib/surveys/inline-survey/inline-survey.ts

```diff
@@ -0,0 +1,117 @@
+import { InvalidQuerySelectorException } from '../../core/exceptions/invalid-query-selector.exception';
+import { UrlFactory } from '../../core/factories/url.factory';
+import { SurveyConfigBuilder } from '../../survey-config-builder/survey-config.builder';
+
+import { InlineSurveyConfig } from './inline-survey-config.interface';
+import { InlineSurveyConfigValidator } from './inline-survey.config-validator';
+
+/**
+ * Class creates iframe element in container referenced by query selector,
+ * it parses and validate provided configuration object
+ *
+ * ### Example (es module)
+ * ```js
+ * import { SurveyConfigBuilder, InlineSurvey } from 'hc.website-sdk'
+ * const configBuilder = new SurveyConfigBuilder({
+ *   baseUrl: 'https://base.com/route',
+ *   language: 'EN',
+ *   extra: {
+ *     isPreview: true
+ *   }
+ * });
+ * const inlineSurvey = new InlineSurvey(configBuilder, {
+ *   elementSelector: '#survey'
+ * });
+ * ```
+ *
+ * ### Example (commonjs)
+ * ```js
+ * var survey = require('hc.website-sdk');
+ * var configBuilder = new survey.SurveyConfigBuilder({
+ *   baseUrl: 'https://base.com/route',
+ *   language: 'EN',
+ *   extra: {
+ *     isPreview: true
+ *   }
+ * });
+ * var inlineSurvey = new survey.InlineSurvey(configBuilder, {
+ *   elementSelector: '#survey'
+ * });
+ *
+ * ```
+ */
+export class InlineSurvey {
+  private readonly iFrameHandle: HTMLIFrameElement;
+  private readonly validator: InlineSurveyConfigValidator;
+  private readonly urlFactory: UrlFactory;
+
+  constructor(
+    configBuilder: SurveyConfigBuilder,
+    private inlineConfig: InlineSurveyConfig
+  ) {
+    this.urlFactory = configBuilder.getUrlFactory();
+    this.validator = new InlineSurveyConfigValidator();
+    this.validator.validateAndThrowOnErrors(inlineConfig);
+    this.iFrameHandle = this.createIframeElement();
+    this.reload();
+  }
+
+  /**
+   * Get iframe element displaying embedded survey
+   */
+  public get iFrame(): HTMLIFrameElement {
+    return this.iFrameHandle;
+  }
+
+  /**
+   * Show survey
+   */
+  public show(): void {
+    this.iFrameHandle.style.display = '';
+  }
+
+  /**
+   * Hide survey
+   */
+  public hide(): void {
+    this.iFrameHandle.style.display = 'none';
+  }
+
+  /**
+   * Reload survey iframe with url produced bu UrlFactory
+   */
+  public reload(): void {
+    this.iFrameHandle.src = this.urlFactory.getUrlWithParams();
+  }
+
+  /**
+   * Creates iframe element inside parent container and returns reference to it
+   *
+   * @throws [[InvalidQuerySelectorException]]
+   * @private
+   */
+  private createIframeElement(): HTMLIFrameElement {
```

Comment:
I could be useful for the devs to get also "destroy" method implemented (to clean up and remove iframe from the dom).